gig/views.py

```diff
@@ -195,28 +211,40 @@ def form_valid(self, form):
 
 class DuplicateView(CreateView):
 
+    original_gig = None
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.original_gig = get_object_or_404(Gig, id=self.kwargs['pk'])
+
     def test_func(self):
         if not self.request.user.is_authenticated:
             return self.handle_no_permission()
         gig = get_object_or_404(Gig, id=self.kwargs['pk'])
         return gig.band.is_editor(self.request.user)
 
+    def get_context_data(self, **kwargs):
+        self.original_gig = Gig.objects.get(id=self.kwargs['pk'])
+        context = super().get_context_data(**kwargs)
+
+        SetupFormTimes(context, self.original_gig)
+        return context
+
     def get_form_kwargs(self, *args, **kwargs):
         kwargs = super().get_form_kwargs(*args, **kwargs)
-        gig_orig = Gig.objects.get(id=self.kwargs['pk'])
         # we didn't originally get a band in the request, we got a gig pk - so get the band from that
         # and stash it among the args from the request
-        self.kwargs['bk'] = gig_orig.band.id
+        self.kwargs['bk'] = self.original_gig.band.id
+
 
         # populate the initial data from the original gig
-        kwargs['initial'] = forms.models.model_to_dict(gig_orig)
+        kwargs['initial'] = forms.models.model_to_dict(self.original_gig)
         # ...but replace the title with a 'copy of'
         kwargs['initial']['title'] = f'Copy of {kwargs["initial"]["title"]}'
         return kwargs
 
     def get_band_from_kwargs(self, **kwargs):
-        # didn't have the band from the request args, so pull it from the gig
-        return get_object_or_404(Gig, id=self.kwargs['pk']).band
+        return self.original_gig.band
```

Comment:
It seems like this method's behavior no longer matches its name. Do we even still need it?

Reply:
We do need it but the name could be better. Will fix.